OpenRA.Mods.Common/Traits/LinkClientBase.cs

```diff
@@ -0,0 +1,54 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public abstract class LinkClientBaseInfo : ConditionalTraitInfo, ILinkClientInfo, Requires<ILinkClientManagerInfo> { }
+
+	public abstract class LinkClientBase<InfoType> : ConditionalTrait<InfoType>, ILinkClient, INotifyCreated where InfoType : LinkClientBaseInfo
+	{
+		readonly Actor self;
+
+		public abstract BitSet<LinkType> GetLinkType { get; }
+		public LinkClientManager LinkClientManager { get; }
+
+		protected LinkClientBase(Actor self, InfoType info)
+			: base(info)
+		{
+			this.self = self;
+			LinkClientManager = self.TraitOrDefault<LinkClientManager>();
```

Comment:
```suggestion
			LinkClientManager = self.Trait<LinkClientManager>();
```

since we require it?

Reply:
we do not require it

8e7fa267092594ef6e1d5255a5d0f2722f249dbb